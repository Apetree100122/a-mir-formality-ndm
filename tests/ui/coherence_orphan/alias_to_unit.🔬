[
    crate core {
        trait CoreTrait<> where [] {}

        trait Unit<> where [] {
            type Assoc<> : [] where [];
        }

        impl<ty T> Unit<> for T where [] {
            type Assoc<> = () where [];
        }
    },
    crate foo {
        struct FooStruct<> where [] {}
        impl<> CoreTrait<> for <FooStruct<> as Unit<>>::Assoc<> where [] {}
    }
]
