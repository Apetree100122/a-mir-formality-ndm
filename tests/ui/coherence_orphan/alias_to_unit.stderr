Error: orphan_check(impl CoreTrait for <FooStruct as Unit>::Assoc where [] { })

Caused by:
    failed to prove {@ IsLocal(CoreTrait(<FooStruct as Unit>::Assoc))} given {}, got {}
